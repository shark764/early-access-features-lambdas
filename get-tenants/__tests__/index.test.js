const axios = require('axios');

jest.mock('axios');

global.process.env = {
  AWS_REGION: 'us-east-1',
  ENVIRONMENT: 'dev',
  DOMAIN: 'domain',
};

axios.mockImplementation(() => ({
  data: {
    result: [
      {
        name: 'Mock Tenant 1',
        id: '00000000-0000-0000-0000-000000000001',
      },
      {
        name: 'Mock Tenant 2',
        id: '00000000-0000-0000-0000-000000000002',
      },
    ],
  },
}));

const mockGet = jest.fn()
  .mockImplementation(() => ({
    promise: () => ({
      Body: '',
    }),
  }));

const mockGetSecretValue = jest.fn()
  .mockImplementation(() => ({
    promise: () => ({
      SecretString: JSON.stringify({
        id: 'mock-secret-id',
        secret: 'mock-secret',
      }),
    }),
  }));

const mockSendMessage = jest.fn()
  .mockImplementation(() => ({
    promise: () => ({}),
  }));

const mockGetQueueUrl = jest.fn()
  .mockImplementation(() => ({
    promise: () => ({
      QueueUrl: 'mock-beta-features-queue-url',
    }),
  }));

jest.mock('aws-sdk', () => ({
  config: {
    update: jest.fn(),
  },
  SQS: jest.fn().mockImplementation(() => ({
    getQueueUrl: mockGetQueueUrl,
    sendMessage: mockSendMessage,
  })),
  DynamoDB: {
    DocumentClient: jest.fn().mockImplementation(() => ({
      get: mockGet,
    })),
  },
  SecretsManager: jest.fn().mockImplementation(() => ({
    getSecretValue: mockGetSecretValue,
  })),
}));

const index = require('../index');

const { handler } = index;

const event = {
  feature: 'mock-feature',
  featureActiveFlag: true,
};

describe('get-tenants', () => {
  let result;
  beforeAll(async () => {
    result = await handler(event);
  });
  it('returns when the code runs without any error', async () => {
    expect(result).toEqual('Finished');
  });
  it('calls get secret value as expected', () => {
    expect(mockGetSecretValue.mock.calls).toMatchSnapshot();
  });
  it('calls get tenants as expected', () => {
    expect(axios.mock.calls).toMatchSnapshot();
  });
  it('puts events to update-beta-features queue as expected', () => {
    expect(mockSendMessage.mock.calls).toMatchSnapshot();
  });
});
