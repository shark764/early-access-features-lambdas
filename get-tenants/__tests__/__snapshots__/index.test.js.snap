// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-tenants calls get secret value as expected 1`] = `
Array [
  Array [
    Object {
      "SecretId": "us-east-1-dev-smooch-cx",
    },
  ],
]
`;

exports[`get-tenants calls get tenants as expected 1`] = `
Array [
  Array [
    Object {
      "auth": Object {
        "id": "mock-secret-id",
        "secret": "mock-secret",
      },
      "method": "get",
      "url": "https://us-east-1-dev-edge.domain/v1/tenants?regionId=00000000-0000-0000-0000-0000000000",
    },
  ],
]
`;

exports[`get-tenants puts events to update-beta-features queue as expected 1`] = `
Array [
  Array [
    Object {
      "MessageBody": "{\\"tenantId\\":\\"00000000-0000-0000-0000-000000000001\\",\\"feature\\":\\"mock-feature\\",\\"featureActiveFlag\\":true}",
      "QueueUrl": "mock-beta-features-queue-url",
    },
  ],
  Array [
    Object {
      "MessageBody": "{\\"tenantId\\":\\"00000000-0000-0000-0000-000000000002\\",\\"feature\\":\\"mock-feature\\",\\"featureActiveFlag\\":true}",
      "QueueUrl": "mock-beta-features-queue-url",
    },
  ],
]
`;
